src/common/data-access/MessagesRepository.ts

```diff
@@ -0,0 +1,10 @@
+import MessageCollection from '@/datamodel/MessageCollection';
+
+export default abstract class MessageRepository {
```

Comment:
In general I agree, but it's there to make obvious that the MessageRepository needs a list of MessageKeys when it is constructed. To avoid using tow interfaces ( see: https://stackoverflow.com/questions/13407036/how-does-typescript-interfaces-with-construct-signatures-work ) and write code around it to make it work, I thought a abstract class is the better approach.

Reply:
I don't think we should misuse an abstract class in order to be able to specify the constructor's signature. The fact that objects of this type have the key lists available at construction time seems like an implementation detail to me.

If ever, I think the relevant method `getMessagesInLanguage` could do with a comment stating how it's intended to be used and that it should return a collection of all required messages.